client/components/data/query-site-comments/index.jsx

```diff
@@ -3,23 +3,37 @@
  */
 import { Component } from 'react';
 import { connect } from 'react-redux';
+import { each, isEmpty } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { requestCommentsList } from 'state/comments/actions';
+import { requestComment, requestCommentsTree } from 'state/comments/actions';
+import getSiteCommentsTree from 'state/selectors/get-site-comments-tree';
 
-const requestComments = ( { requestSiteComments, siteId, status } ) => (
-	siteId && requestSiteComments( { siteId, status } )
+const requestTree = ( { requestSiteCommentsTree, siteId, status } ) => (
+	siteId && requestSiteCommentsTree( siteId, status )
 );
 
+const requestComments = ( { commentsTree, requestSingleComment, siteId } ) => {
+	if ( isEmpty( commentsTree ) ) {
+		return;
+	}
+	each( commentsTree, ( { commentId } ) => {
+		requestSingleComment( siteId, commentId );
+	} );
```

Comment:
Careful when testing, this currently fetches every single comment on your site that matches the current filter. We'll want to update this to fetch the page we're on currently.